server/src/main/java/org/opensearch/index/fielddata/ordinals/MultiOrdinals.java

```diff
@@ -213,13 +214,14 @@ public long getValueCount() {
         public boolean advanceExact(int docId) throws IOException {
             currentOffset = docId != 0 ? endOffsets.get(docId - 1) : 0;
             currentEndOffset = endOffsets.get(docId);
-            return currentOffset != currentEndOffset;
+            docValueCount = Math.toIntExact(currentEndOffset - currentOffset);
+            return docValueCount > 0;
         }
 
         @Override
         public long nextOrd() throws IOException {
             if (currentOffset == currentEndOffset) {
-                return SortedSetDocValues.NO_MORE_DOCS;
+                throw new IllegalStateException("nextOrd() called more than docValueCount() times");
```

Comment:
I am wondering if this exception and `Called nextOrd after docValueCount` have different meaning? If not, we should reuse the same message for all instances? Should we create custom exception for this case?

Reply:
The `Called nextOrd after docValueCount` was something generated by IntelliJ's auto-complete. This message is better IMO. I can make it consistent across the different implementations.

In practice, these checks should be short-lived. Lucene doesn't even do checks. The JavaDoc there just says "It is illegal to call this more than docValueCount() times for the currently-positioned doc." Some implementations will just return the same value over and over. Other implementations may do something else.

I added these exceptions to highlight the places where we were doing the wrong thing. The later commits on this PR were where I fixed those places. In theory, before we ship OpenSearch 3.0, we could probably remove these exceptions.

Reply:
Sure, let us make these exceptions consistent.